### PR TITLE
make Cython wrapper for the C++ library

### PR DESCRIPTION
**Pull request by anthrotype**
I managed to write a minimalist Cython wrapper for compreffor's C++ library.
I also made the setup.py build the extension module without needing any `make`, but just using the usual distutils/setuptools.

Note that Cython itself is not required to build from a git checkout, as I am also including the pre-generated `_compreffor.cpp` source file. If the developer has cython installed (>=0.24) then it uses that to re-generate the cpp source file from the pyx file.

Of course, a modern C++ compiler (with support for C++11) is still required (cf. #11).

The Cython-generated native extension module is faster than both the old methods to wrap the C++ code (i.e. executable via subprocess and shared library via ctypes). So I removed the latter two methods, and only left as the pure-python method and the new Python/C++ extension one.

### CODE REVIEW

File: compreffor/__init__.py [hunk 1]
```diff
@@ -42,26 +42,20 @@
                              to run
 
 Compression Backends:
-There are 3 different ways the compreffor can be run.
+There are 2 different ways the compreffor can be run.
     - First is a pure python approach, which can be selected from this module
       by passing method=Methods.Py. This backend is significantly slower than
       the other 2 backends (~10-20x). The logic for this backend can be found
```
Comment (khaledhosny) [inline compreffor/__init__.py]: Should be `than the other backend` now.

File: compreffor/__init__.py [hunk 2]
```diff
@@ -42,26 +42,20 @@
...
-      prepared in python and then the executable `cffCompressor` is started
-      and data piped in. It's response is read back in over another pipe and
-      python writes the result. This can be selected here using
-      Methods.CxxExecutable and logic can be found in cxxCompressor.py,
-      cffCompressor.h, and cffCompressor.cc.
-    - Third is C++ dynamic library (shared object) backed. With this method,
+    - Second is C++ extension module backed. With this method,
       python calls the relevant functions directly from `libcompreff.so` rather
       than starting an executable in another process. This is currently slower
       than the second approach. This can be selected here using Methods.CxxLib
```
Comment (khaledhosny) [inline compreffor/__init__.py]: I think the last paragraph needs to be reworded as well.

**Comment by anthrotype** [comment]
@khaledhosny thanks for having a look. I applied the changes you suggested.

**Comment by anthrotype** [comment]
ping anyone?

**Comment by behdad** [comment]
Thanks @anthrotype!

**Pull request merged**